docs/reference/cli-config.rst

```diff
@@ -0,0 +1,39 @@
+Command Line Interface Configuration
+====================================
+
+The Testflinger CLI can be configured on the command line using the parameters described in ``testflinger --help``. However, sometimes it also supports using environment variables, or reading values from a configuration file. This can be helpful for CI/CD environments, or for setting up a config with different default values read from a config file.
+
+The configuration file is read from ``$XDG_CONFIG_HOME/testflinger-cli.conf`` by default, but this can be overridden with the ``--configfile`` parameter.
```

Comment:
Question: does every sub-command of testflinger-cli support this parameter? If used, could you specify if the conf file is used for the current session or only for this single command?

Reply:
It would only alter the config file that is read for the current execution - not permanently. I think that's typical for commands that take a parameter to specify where to read the config file though. Do you have a suggestion for how I could phrase this better?

Reply:
Indeed it's a typical usage, but since we are referring to multiple override orders, it might be clearer if we list the resolution order altogether. Trying to add a new paragraph here, please correct if it's wrong:

```suggestion
The configuration file is read from ``$XDG_CONFIG_HOME/testflinger-cli.conf`` by default, but this can be overridden with the ``--configfile`` parameter.

When a configuration variable is defined in multiple locations, Testflinger resolves the value by applying the following priority order, from highest to lowest:

1. command-line parameter (for example, ``--server <local_server>``)
2. configuration file specified via the ``--configfile`` parameter
3. environment variable
4. default configuration file located at  ``$XDG_CONFIG_HOME/testflinger-cli.conf``

```